lib/_http_client.js

```diff
@@ -87,6 +87,14 @@ function ClientRequest(options, cb) {
     }
     if (host && !this.getHeader('host') && setHost) {
       var hostHeader = host;
+
+      // For the Host header, ensure that IPv6 addresses are enclosed
+      // in square brackets, as defined by URI formatting
+      // https://tools.ietf.org/html/rfc3986#section-3.2.2
+      if (net.isIPv6(hostHeader)) {
```

Comment:
There were no check for correctness here before, so I assume that there are two possibilities:

1. Host is check before and is valid at this point, so if there is `:` then it can only possibly be IPv6 address.

2. Host isn't checked before, so it might be invalid. If it has `:` then it is either IPv6 or invalid. If we wrap invalid host in brackets it would still be invalid, so it doesn't really matter.

`isIPv6` involves calling C++ code and also checking if argument is IPv4 before checking if it's IPv6.